addons/hr_work_entry/models/hr_work_entry.py

```diff
@@ -51,6 +51,12 @@ class HrWorkEntry(models.Model):
     # FROM 7s by query to 2ms (with 2.6 millions entries)
     _contract_date_start_stop_idx = models.Index("(version_id, date) WHERE state IN ('draft', 'validated')")
 
+    @api.constrains('duration')
+    def _check_duration(self):
+        for work_entry in self:
+            if work_entry.duration <= 0 or work_entry.duration > 24:
+                raise ValidationError(self.env._("Duration must be positive and less than 24 hours."))
```

Comment:
```suggestion
                raise ValidationError(self.env._("Duration must be positive and at most 24 hours."))
```
Wouldn't this be more expressive of the code's behavior?